sql/core/src/test/scala/org/apache/spark/sql/connector/WriteDistributionAndOrderingSuite.scala

```diff
@@ -569,4 +785,26 @@ class WriteDistributionAndOrderingSuite
         fail("expected V2TableWriteExec")
     }
   }
+
+  private def orderedWritePartitioning(
+      writeOrdering: Seq[catalyst.expressions.SortOrder],
+      targetNumPartitions: Option[Int]): physical.Partitioning = {
+    targetNumPartitions match {
```

Comment:
it's simply `RangePartitioning(writeOrdering, targetNumPartitions.getOrElse(conf.numShufflePartitions))`